.github/workflows/build-macos.yaml

```diff
@@ -60,10 +60,10 @@ jobs:
         run: |
           # The goal here is for $BUILD_COMMIT to be the HEAD (necessary for
           # git-annex's version detection to use the correct git commit) with
-          # the contents of master — or whatever ref is being used as the
-          # workflow source — also available.
+          # the .github/ and patches/ trees from master — or whatever ref is
+          # being used as the workflow source — also available.
           git checkout "$BUILD_COMMIT"
-          git checkout "$GITHUB_SHA" -- .
+          git checkout "$GITHUB_SHA" -- .github patches
```

Comment:
so what version did it build before then??

Reply:
This change just restricts the checkout of `master` to only include the `.github/` and `patches/` directories.  Previously, all files on master were checked out, but now that we're adding more things at the top level, it's better to only fetch what's actually needed in order to avoid a chance of collisions.